docs/null_safety_migration.md

```diff
@@ -0,0 +1,166 @@
+# Migrating a Builder to support null safety
+
+The null safety feature in Dart presents several challenges for Builder
+authors, this is a living document which will address some of the core issues,
+as well as potential solutions.
+
+## Language versioning
+
+Language versioning is a new feature introduced to allow an incremental
```

Comment:
Are there other language versioning docs we should link to?

Reply:
cc @leafpetersen do we have a canonical doc?

Reply:
Perhaps not yet. https://github.com/dart-lang/site-www/issues/2554

Reply:
There is a feature specification for this in the language repository, but I don't know about anything broader than that.

Reply:
Ya I don't really want to link to the full spec here, although we could. I am also concerned it will move around and don't want to have to keep the link up to date (it is currently under `accepted/future-releases`).

Reply:
Going to merge as is for now and we can add a more canonical user facing link when we have it.